mobile/app/(dashboard)/reservations.tsx

```diff
@@ -15,20 +18,27 @@ export default function ReservationsPage() {
 
     const { removeReservation } = useRemoveReservation();
 
-    const handleRemoveReservation = async (reservationId: number) => {
-        try {
-            await removeReservation(reservationId);
-            await refetch();
-            toast.success("Reservation removed successfully.");
-        } catch (error) {
-            const errorMessage = (error as { message?: string }).message;
-
-            toast.error("Failed to remove reservation", {
-                description: errorMessage,
-            });
-        }
+    const handleRemoveReservation = (reservationId: number) => {
+        withConfirmation("Are you sure you want to remove this reservation?", async () => {
+            try {
+                await removeReservation(reservationId);
+                await refetch();
+                toast.success("Reservation removed successfully.");
+            } catch (error) {
+                const errorMessage = (error as { message?: string }).message;
+                toast.error("Failed to remove reservation", {
+                    description: errorMessage,
+                });
+            }
+        });
     };
 
+    useFocusEffect(
+        useCallback(() => {
+            refetch();
+        }, [refetch])
+    );
```

Comment:
Usually memoization (`useCallback`, `useMemo`, `memo`) is not necessary at all, idk this `useFocusEffect` hook, but I suppose it should work exactly the same even if `refetch` is not memoized. Performance-wise it can be even better to not memoize such things, as memoization can be more expensive than recalculation of memoized object. And it's more readable.

Over-usage of memoization is one of most common issues in React. It can stay like this, it doesn't really make any difference in our case, I am just pointing this out so you don't need to worry about adding those `useCallbacks` etc

```suggestion
    useFocusEffect(refetch);
```